pkg/agent/containerd/watcher.go

```diff
@@ -36,7 +36,7 @@ func CreateWatcher() (*Watcher, error) {
 	return &Watcher{
 		watcher:   watcher,
 		filesMap:  make(map[string]fs.FileInfo),
-		workqueue: workqueue.NewTypedRateLimitingQueue(workqueue.DefaultTypedControllerRateLimiter[fsnotify.Event]()),
+		workqueue: workqueue.TypedNewDelayingQueue[fsnotify.Event](),
```

Comment:
Can you confirm that this works properly?

I think yu need to use the same key for each entry for this to do what you want. If you do it like this, each event will get its own entry in the queue. While events do implement Comparable, different events for the same filename are NOT the the same event and will not (as far as I know) compare as equal.

What you want is to use `event.Name` as the key, so that additional events with the same name will further delay processing of the workqueue item. That's why I suggested using DelayingQueue, which takes string as the key.

Reply:
Setting a workqueue with `event.Name` as the value, will work with the events for the same file, but, it generates a complexity that we will need to have something to cache the event, I created a event cache, its a map and can be access inside the workqueue, but it does not feel right, maybe I could be missing something in the workqueue. Do you have any suggestion?